1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/bedwars/BedwarsMod.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright © 2021-2023 moehreag <moehreag@gmail.com> & Contributors
+ *
+ * This file is part of AxolotlClient.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * For more information, see the LICENSE file.
+ */
+
+package io.github.axolotlclient.modules.hypixel.bedwars;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import io.github.axolotlclient.AxolotlClientConfig.options.BooleanOption;
+import io.github.axolotlclient.AxolotlClientConfig.options.EnumOption;
+import io.github.axolotlclient.AxolotlClientConfig.options.OptionCategory;
+import io.github.axolotlclient.modules.hypixel.AbstractHypixelMod;
+import io.github.axolotlclient.util.events.Events;
+import io.github.axolotlclient.util.events.impl.ReceiveChatMessageEvent;
+import io.github.axolotlclient.util.events.impl.ScoreboardRenderEvent;
+import io.github.axolotlclient.util.events.impl.WorldLoadEvent;
+import lombok.Getter;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.PlayerListEntry;
+import net.minecraft.scoreboard.Scoreboard;
+import net.minecraft.scoreboard.ScoreboardPlayerScore;
+import net.minecraft.scoreboard.Team;
+import net.minecraft.text.LiteralText;
+import net.minecraft.util.Formatting;
+
+/**
+ * @author DarkKronicle
+ */
+
+public class BedwarsMod implements AbstractHypixelMod {
+
+	private final static Pattern[] GAME_START = {
+		Pattern.compile("^\\s*?Protect your bed and destroy the enemy beds\\.\\s*?$"),
+		Pattern.compile("^\\s*?Bed Wars Lucky Blocks\\s*?$"),
+		Pattern.compile("^\\s*?Bed Wars Swappage\\s*?$")
```

Comment:
Have to figure out which dreams mode the mod should initiate in. Swappage should almost certainly be removed since team swaps will completely break the mod. This was originally added just to provide basic testing.

Reply:
I'd assume that all modes work except ones where teams are mixed up (which means just Swappage currently)

Reply:
I think only swappage and lucky blocks won't work. Lucky blocks has a placeable bed.

Reply:
true, I forgot about that.